head/src/main/kotlin/io/qalipsis/core/head/report/thymeleaf/MediaReplacedElementFactory.kt

```diff
@@ -72,4 +73,42 @@ internal class MediaReplacedElementFactory(private val superFactory: ReplacedEle
 
     override fun setFormSubmissionListener(listener: FormSubmissionListener?) =
         superFactory.setFormSubmissionListener(listener)
+
+    /**
+     * Converts the svg image to a convenient png byte array format.
+     *
+     * @param imageSrc the image string content
+     */
+    private fun svgToPng(imageSrc: String): ByteArray {
+        val base64ByteArray = Base64.getDecoder().decode(imageSrc)
+        // Set the input source as the SVG data.
+        val input = TranscoderInput(ByteArrayInputStream(base64ByteArray))
+        // Create an output stream for the converted image.
+        return ByteArrayOutputStream().use { outputStream ->
+            val output = TranscoderOutput(outputStream)
```

Comment:
Do not create variable when not necessary, directly pass TranscoderOutput(outputStream) as argument of transcoder.transcode. Same for the input (if use.{} is not necessary on it, see my comment below)